Modules/_datetimemodule.c

```diff
@@ -6782,70 +6817,67 @@ _datetime_exec(PyObject *module)
         return -1;
     }
 
-    PyObject *x = create_timezone(delta, NULL);
+    datetime_state *st = STATIC_STATE();
+    st->utc = create_timezone(delta, NULL);
     Py_DECREF(delta);
-    if (x == NULL) {
-        return -1;
+    if (st->utc == NULL) {
+        goto error;
```

Comment:
Apparently, now exec function has two different parts:

* First part uses return -1
* Second part initializes the static state and uses goto to make sure that the state always stays in a consistent state: either fully initialized, or cleared

I suggest to split the function in two parts (create a subfunction). I don't think that a second PR is needed.

Maybe just add a second _datetime_init_state() function?

Reply:
Sure, we can bake it into this PR. I'll get to it later tonight.